Investigations/SUPPORT_Notebook.ipynb

```diff
@@ -0,0 +1,440 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#%% -------- Plot Histograms For All The Variable Distributions -------- #\n",
+    "\n",
+    "plot_variable_distributions(\n",
+    "    categorical_columns=original_categorical_columns,\n",
+    "    continuous_columns=original_continuous_columns,\n",
+    "    data_supp=data_supp,\n",
+    "    synthetic_supp=synthetic_supp,\n",
+    "    saving_filepath=\"\",\n",
+    "    pre_proc_method=pre_proc_method,\n",
+    ")\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "09630b17",
+   "metadata": {},
+   "source": [
+    "## Metric evaluation\n",
+    "\n",
+    "We use the SDV evaluation framework. Supply the metrics you wish to find in the user_metrics list from SDV guidance. Can start here: https://sdv.dev/SDV/user_guides/evaluation/single_table_metrics.html\n",
+    "\n",
+    "Note that not all of these will work, some are hit and miss. We predominantly rely on continuous and discrete KL divergence measures. You can also input <b>\"gower\"</b> and this will calculate the gower distance using the gower library."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0f240e21",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#%% -------- SDV Metrics -------- #\n",
+    "\n",
+    "# Define the metrics you want the model to evaluate\n",
+    "\n",
+    "gower = False\n",
+    "\n",
+    "metrics = distribution_metrics(\n",
+    "    gower_bool=gower,\n",
+    "    data_supp=data_supp,\n",
+    "    synthetic_supp=synthetic_supp,\n",
+    "    categorical_columns=original_categorical_columns,\n",
+    "    continuous_columns=original_continuous_columns,\n",
+    "    saving_filepath=None,\n",
+    "    pre_proc_method=pre_proc_method,\n",
+    ")\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ecc9065d",
+   "metadata": {},
+   "source": [
+    "# Privacy Metric Evaluation\n",
+    "\n",
+    "Using SDV privacy metrics we can get an insight into how privacy is conserved when utilising dp-sgd methods. SDV's privacy metrics are limited in that they can only be used on similar data types. E.g. if we choose age to be the sensitive variably, we can build ML based models to predict a users age using the other columns. However we are forced to only use columns that are also continuous variables."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "de25de2b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Specify our private variable\n",
+    "\n",
+    "private_variable = \"x14\"\n",
+    "\n",
+    "privacy_metric = privacy_metrics(\n",
+    "    private_variable=private_variable,\n",
+    "    data_supp=data_supp,\n",
+    "    synthetic_supp=synthetic_supp,\n",
+    "    categorical_columns=original_categorical_columns,\n",
+    "    continuous_columns=original_continuous_columns,\n",
+    "    saving_filepath=None,\n",
+    "    pre_proc_method=pre_proc_method,\n",
+    ")\n",
```

Comment:
Possibly add a print of the metrics here to show the outputs in the notebook as well